### DISCUSSION TITLE
Reference projects

### DISCUSSION THREAD

**Discussion by mmamayka** [BESM-Consortium/BESM-666#8] [Sources]
In this discussion I would introduce some projects, documentation files & etc., where we can get functional simulator implementation ideas & references.

**Comment by mmamayka** [comment]
- [Unicorn](https://github.com/unicorn-engine/unicorn) is a QEMU based functional simulator supporting RISCV architecture
- [MIPT-MIPS](https://mipt-ilab.github.io/mipt-mips/) MIPT & Intel lection course abount CPU uArch & code development & testing
- [MIPT-MIPS-CEN64](https://github.com/MIPT-ILab/cen64) MIPS uArch simulator developed for MIPT-MIPS course

**Reply by mmamayka** [reply]
[Habr article "RISCV from zero"](https://habr.com/ru/articles/454208/)

**Comment by mmamayka** [comment]
[RISCV ISA specification](https://riscv.org/technical/specifications/)
[RISCV calling convention](https://riscv.org/wp-content/uploads/2015/01/riscv-calling.pdf)

**Comment by mmamayka** [comment]
[ELF Specification](https://refspecs.linuxfoundation.org/elf/elf.pdf)

**Comment by c71n93** [comment]
[ELFIO](https://github.com/serge1/ELFIO) - library for reading and generating files in the ELF binary format
[ELFIO manual](https://elfio.sourceforge.net/elfio.pdf)
